src/authextension.ts

```diff
@@ -15,29 +18,70 @@ export module authextension {
     process.env['THREE_SCALE_USER_TOKEN'] = context_f8_3scale_user_key;
   };
 
-  export const authorize_f8_analytics = context => {
-    return new Promise((resolve, reject) => {
+  setUUID = (uuid) => {
+    process.env['UUID'] = uuid;
+  };
+
+  export const authorize_f8_analytics = async context => {
+    try {
       let context_f8_access_routes = context.globalState.get(
         'f8_access_routes'
       );
       let context_f8_3scale_user_key = context.globalState.get(
         'f8_3scale_user_key'
       );
+
       if (context_f8_access_routes && context_f8_3scale_user_key) {
         setContextData(context_f8_access_routes, context_f8_3scale_user_key);
-        resolve(true);
       } else {
-        get_3scale_routes(context)
-          .then(resp => {
-            resolve(resp);
-          })
-          .catch(err => {
-            reject(err);
-          });
+        let respData = await get_3scale_routes(context);
+        if (!respData) {
+          return false;
+        }
       }
-    });
+
+      let uuid = context.globalState.get(GlobalState.UUID);
+
+      if (uuid && uuid != '') {
+        setUUID(uuid);
+      } else {
+        let response = await getUUID(context);
```

Comment:
getUUID shall return the uuid instead of response., that is what the function name conveys.